pandas/tests/indexes/timedeltas/test_indexing.py

```diff
@@ -41,6 +41,14 @@ def test_getitem(self):
             tm.assert_index_equal(result, expected)
             assert result.freq == expected.freq
 
+    def test_timestamp_invalid_key(self):
+        # GH#20464
+        tdi = pd.timedelta_range(0, periods=10)
```

Comment:
do we have a test that indexes with NaT? (both Timedelta and Datetime dtype)

Reply:
We have one for timedelta, just added one for datetime